### PR TITLE
Tasks 461 462 463 464

### PR DESCRIPTION
**Pull request by ThanhNIT**

### CODE REVIEW

**Comment by javadev** [comment]
IMG_20211225_041303

It looks like test coverage is too low.

File: src/main/java/g0401_0500/s0462_minimum_moves_to_equal_array_elements_ii/Solution.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package g0401_0500.s0462_minimum_moves_to_equal_array_elements_ii;
+
+import java.util.Arrays;
```
Comment (javadev) [inline src/main/java/g0401_0500/s0462_minimum_moves_to_equal_array_elements_ii/Solution.java:5]: Please add comment with tags.

**Comment by javadev** [comment]
👌

File: src/main/java/g0401_0500/s0464_can_i_win/Solution.java [hunk 2]
```diff
@@ -22,16 +22,12 @@ private boolean canWin(int state, Boolean[] dp, int desiredTotal, int maxChoosab
...
-                // i is not used
-                // set i as used in state
-                if (i >= desiredTotal
-                        || !canWin(state | cur, dp, desiredTotal - i, maxChoosableInteger)) {
-                    // i is greater than the desired total
-                    // or the other player cannot win after the current player picks i
-                    dp[state] = true;
-                    return dp[state];
-                }
+            if ((cur & state) == 0 && i >= desiredTotal
```
Comment (javadev) [inline src/main/java/g0401_0500/s0464_can_i_win/Solution.java:25]: We need more brackets here ()

File: src/main/java/g0401_0500/s0464_can_i_win/Solution.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        // state is the bitmap representation of the current state of choosable integers left
+        // dp[state] represents whether the current player can win the game at state
+        if (dp[state] != null) {
+            return dp[state];
+        }
+        for (int i = 1; i <= maxChoosableInteger; i++) {
+            // current number to pick
+            int cur = 1 << (i - 1);
+            if ((cur & state) == 0 && i >= desiredTotal
+                    || !canWin(state | cur, dp, desiredTotal - i, maxChoosableInteger)) {
```
Comment (ThanhNIT) [inline src/main/java/g0401_0500/s0464_can_i_win/Solution.java:26]: @javadev  because we've entered a newline, you can't see the brackets

**Pull request merged**